includes/data-stores/class-wc-admin-reports-customers-data-store.php

```diff
@@ -508,6 +536,26 @@ public static function get_customer_id_by_user_id( $user_id ) {
 		return $customer_id ? (int) $customer_id : false;
 	}
 
+	/**
+	 * Retrieve a customer id by billing email.
+	 *
+	 * @param string $email Billing email for the user.
+	 * @return false|int Customer ID if found, boolean false if not.
+	 */
+	public static function get_customer_id_by_email( $email ) {
+		global $wpdb;
+
+		$table_name  = $wpdb->prefix . self::TABLE_NAME;
+		$customer_id = $wpdb->get_var(
+			$wpdb->prepare(
+				"SELECT customer_id FROM {$table_name} WHERE email = %s AND user_id IS NULL LIMIT 1",
```

Comment:
It looks like this is only searching for customers without a `user_id` (guests only).  Is there a difference between this function and `get_guest_by_email()`?

I think we can remove it if we go with the above refactor, but if I'm wrong we might want to rename this to something like `get_guest_customer_id_by_email()`.

Reply:
I agree it's not the nicest solution. I did what you say initially, but then I noticed we need a static method, so I created the new fn (`get_guest_by_email` casts numbers based on member var `$column_types` , thus can't be static).

Reply:
Maybe we could remove the casting from `get_guest_by_email()` then since only the customer ID is needed anyway.